File: frontend/src/constants/visitor.ts
import type {VisitorPassStatus,VisitorEffectiveState} from '../types';

export const VISITOR_STATUS:Record<Uppercase<VisitorPassStatus>,VisitorPassStatus>={PENDING:'pending',APPROVED:'approved',REJECTED:'rejected',REVOKED:'revoked'};

export const visitorStatusText:Record<VisitorPassStatus,string>={pending:'待审',approved:'已批准',rejected:'已拒绝',revoked:'已撤销'};

export const visitorStateText:Record<VisitorEffectiveState,string>={upcoming:'未生效',active:'通行中',expired:'已过期'};

// 徽标颜色与后端 VisitorPassStatus / EffectiveState 枚举一一对应。
export const visitorStatusTagType:Record<VisitorPassStatus,'warning'|'success'|'danger'|'info'>={pending:'warning',approved:'success',rejected:'danger',revoked:'info'};

export const visitorStateTagType:Record<VisitorEffectiveState,'info'|'success'|'info'>={upcoming:'info',active:'success',expired:'info'};

export const VISITOR_ERROR_CODE_CONFLICT=40901;

File: backend/internal/constants/visitor.go
package constants

// VisitorPassStatus 访客通行证申请状态。
const (
	VisitorStatusPending  = "pending"  // 待审
	VisitorStatusApproved = "approved" // 已批准（有效通行证，过期前有效）
	VisitorStatusRejected = "rejected" // 审批未通过（冲突或人工拒绝），保持/退回待审语义由前端提示
	VisitorStatusRevoked  = "revoked"  // 到访开始前被撤销
)

// VisitorEffectiveState 是审批通过后根据当前时间计算的有效态（不入库）。
const (
	VisitorStateUpcoming = "upcoming" // 未到到访开始时间
	VisitorStateActive   = "active"   // 到访时段内
	VisitorStateExpired  = "expired"  // 已过期，时段已释放
)

var ValidVisitorStatuses = map[string]bool{
	VisitorStatusPending:  true,
	VisitorStatusApproved: true,
	VisitorStatusRejected: true,
	VisitorStatusRevoked:  true,
}

// VisitorPlateMinLen/VisitorPlateMaxLen 车牌长度边界（含新能源 8 位）。
const (
	VisitorPlateMinLen  = 5
	VisitorPlateMaxLen  = 16
	VisitorReasonMaxLen = 200
)
